### ISSUE TITLE
react calc sourcecode (1)

### ISSUE THREAD

**Issue by zzz6519003** [zzz6519003/blog#55]
First you can have a look at the demo project @ http://jsfiddle.net/Snowmanzzz/8xL5zs99/ or http://codepen.io/snowmanzzz/pen/VvEypo
I copied from other guy, cuz think it's more convenient in jsfiddle than in codepen.
```js
const TotalRecall = React.createClass({
  _toggleMemories() {
    this.setState({show: !this.state.show});
  },
  
  _recallMemory(memory) {
    store.newInput = memory;
    ee.emitEvent('toggle-memories');
  },
  
  getInitialState() {
    return {show: false}
  },
  
  componentWillMount() {
    ee.addListener('toggle-memories', this._toggleMemories);
  },
  
  render() {
    let classNames = `memory-bank ${this.state.show ? 'visible' : ''}`;
    
    return (
      <section className={classNames}>
        <Button text="+" clickHandler={this._toggleMemories} klass="toggle-close" />
        {store.curMemories.map((mem) => {
          return (
            <Button klass="block memory transparent" text={mem} clickHandler={this._recallMemory} />
          );
        })}
      </section>
    )
  }
});
```
From this component, we can see `<Button klass="block memory transparent" text={mem} clickHandler={this._recallMemory} />` register a click event on this button, it's passing
```js
  _recallMemory(memory) {
    store.newInput = memory;
    ee.emitEvent('toggle-memories');
  },
```
as a event handler to the Button component as a prop.

So let's turn to Button.
```js
const Button = React.createClass({
	_handleClick() {
		let text = this.props.text,
			cb = this.props.clickHandler;

		if (cb) {
			cb.call(null, text);
		}
	},

	render() {
		return (
			<button className={this.props.klass} onClick={this._handleClick}>
				<span className="title">{this.props.text}</span>
			</button>
		);
	}
})
```
So here you can see me as a Button, I'm playing RPG, and I'm the Button. You are passing that bunch of code to me as the `clickHandler` prop. Now I see you bitch, what I'm doing is that I have a button, in the handler of it, I have handleClick registered, and in it what I'm going to do is that I have your thing, call it with the text prop of my button. In this case we have `store.curMemories` as the text.
basically looked like this

```js
memory: Array[14]
0: 0
1: "3"
2: "3 + "
3: "3 + 3"
4: 6
5: "3 + 3 * "
6: "3 + 3 * 6"
7: 21
8: "3 + 3 * - "
9: "3 + 3 * -"
10: "3 + 3 * "
11: "3 + 3 * 6"
12: "3 + 3 * 66"
13: 201
length: 14
```
You then set the curinput and so on.

From an instinction, when you click the number, what you wanna do in your callback, is to use the thing i'm clicking at(the text, or the title) to manipulate the store.
But here, we make the "get the paremeter process" in the `Button` component, so much so that we can reuse the logic.
